shared/prater/config.yaml

```diff
@@ -1,155 +1,72 @@
+#  - These may be re-assigned to another fork-version later
+#  - Temporarily set to max uint64 value: 2**64 - 1
 
+# Altair
+ALTAIR_FORK_VERSION: 0x01001020
+ALTAIR_FORK_EPOCH: 36660
+# Merge
+MERGE_FORK_VERSION: 0x02001020
+MERGE_FORK_EPOCH: 18446744073709551615
+# Sharding
+SHARDING_FORK_VERSION: 0x03001020
+SHARDING_FORK_EPOCH: 18446744073709551615
 
-# Initial values
-# ---------------------------------------------------------------
-# Prater area code (Vienna)
-GENESIS_FORK_VERSION: 0x00001020
-BLS_WITHDRAWAL_PREFIX: 0x00
+# TBD, 2**32 is a placeholder. Merge transition approach is in active R&D.
+TRANSITION_TOTAL_DIFFICULTY: 4294967296
 
 
 # Time parameters
 # ---------------------------------------------------------------
-# Customized for Prater: 1919188 seconds (Mar-23-2021 02:00:00 PM +UTC)
-GENESIS_DELAY: 1919188
 # 12 seconds
 SECONDS_PER_SLOT: 12
-# 2**0 (= 1) slots 12 seconds
-MIN_ATTESTATION_INCLUSION_DELAY: 1
-# 2**5 (= 32) slots 6.4 minutes
-SLOTS_PER_EPOCH: 32
-# 2**0 (= 1) epochs 6.4 minutes
-MIN_SEED_LOOKAHEAD: 1
-# 2**2 (= 4) epochs 25.6 minutes
-MAX_SEED_LOOKAHEAD: 4
-# 2**6 (= 64) epochs ~6.8 hours
-EPOCHS_PER_ETH1_VOTING_PERIOD: 64
-# 2**13 (= 8,192) slots ~13 hours
-SLOTS_PER_HISTORICAL_ROOT: 8192
+# 14 (estimate from Eth1 mainnet)
+SECONDS_PER_ETH1_BLOCK: 14
 # 2**8 (= 256) epochs ~27 hours
 MIN_VALIDATOR_WITHDRAWABILITY_DELAY: 256
 # 2**8 (= 256) epochs ~27 hours
 SHARD_COMMITTEE_PERIOD: 256
-# 2**2 (= 4) epochs 25.6 minutes
-MIN_EPOCHS_TO_INACTIVITY_PENALTY: 4
-
-
-# State vector lengths
-# ---------------------------------------------------------------
-# 2**16 (= 65,536) epochs ~0.8 years
-EPOCHS_PER_HISTORICAL_VECTOR: 65536
-# 2**13 (= 8,192) epochs ~36 days
-EPOCHS_PER_SLASHINGS_VECTOR: 8192
-# 2**24 (= 16,777,216) historical roots, ~26,131 years
-HISTORICAL_ROOTS_LIMIT: 16777216
-# 2**40 (= 1,099,511,627,776) validator spots
-VALIDATOR_REGISTRY_LIMIT: 1099511627776
-
-
-# Reward and penalty quotients
-# ---------------------------------------------------------------
-# 2**6 (= 64)
-BASE_REWARD_FACTOR: 64
-# 2**9 (= 512)
-WHISTLEBLOWER_REWARD_QUOTIENT: 512
-# 2**3 (= 8)
-PROPOSER_REWARD_QUOTIENT: 8
-# 2**26 (= 67,108,864)
-INACTIVITY_PENALTY_QUOTIENT: 67108864
-# 2**7 (= 128) (lower safety margin at Phase 0 genesis)
-MIN_SLASHING_PENALTY_QUOTIENT: 128
-# 1 (lower safety margin at Phase 0 genesis)
-PROPORTIONAL_SLASHING_MULTIPLIER: 1
+# 2**11 (= 2,048) Eth1 blocks ~8 hours
+ETH1_FOLLOW_DISTANCE: 2048
 
 
-# Max operations per block
+# Validator cycle
 # ---------------------------------------------------------------
+# 2**2 (= 4)
+INACTIVITY_SCORE_BIAS: 4
 # 2**4 (= 16)
-MAX_PROPOSER_SLASHINGS: 16
-# 2**1 (= 2)
-MAX_ATTESTER_SLASHINGS: 2
-# 2**7 (= 128)
-MAX_ATTESTATIONS: 128
-# 2**4 (= 16)
-MAX_DEPOSITS: 16
-# 2**4 (= 16)
-MAX_VOLUNTARY_EXITS: 16
+INACTIVITY_SCORE_RECOVERY_RATE: 16
+# 2**4 * 10**9 (= 16,000,000,000) Gwei
+EJECTION_BALANCE: 16000000000
+# 2**2 (= 4)
+MIN_PER_EPOCH_CHURN_LIMIT: 4
```

Comment:
we don't have a good way to change these constants _in a fork_ - cc @protolambda  - ie we can introduce a new constant but that's messy, but we can really redefine an existing one because it's so tightly tied to spec processing - something to consider

Reply:
yeah, the current method would be to define a new constant coupled to that fork.

for chains you don't want to update, you can either define the constant as the prior value, or leave it undefined and check for absence of the value. I'd think the former to be less error prone

Reply:
The spec processing would need to define when/how a new constant would go into effect. Some constants apply to older epochs pre-fork just after the transition, and every constant is necessary to sync the past chain. It's much less error prone to define changes with a new name. Testnets changing things in a hardfork that mainnet never changes also just means more divergence, which we should avoid.

Reply:
I'm mostly highlighting it for consideration of a potential change in the existing testnets, should we want to catch up with mainnet in terms of validator counts (or whatever) - it's not that "easy".